### PR TITLE
2839 endpoint

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

File: web-api/src/cases/publicSearchLambda.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @returns {Promise<*|undefined>} the api gateway response object containing the statusCode, body, and headers
+ */
+exports.handler = event =>
+  handle(event, async () => {
+    const user = {}; // TODO - need to figure out how this is going to work
+    const applicationContext = createApplicationContext(user);
+    try {
+      const results = await applicationContext
+        .getUseCases()
+        .casePublicSearchInteractor({
```
Comment (rachaelparris) [inline web-api/src/cases/publicSearchLambda.js]: Could you name this file the same as the interactor - `casePublicSearchLambda`?

**Review by codyseibert — APPROVED** [review]

File: shared/src/business/useCases/casePublicSearchInteractor.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * casePublicSearchInteractor
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {string} providers.countryType the country type to search cases by (domestic/international)
+ * @param {string} providers.petitionerName the name of the petitioner to search cases by
+ * @param {string} providers.petitionerState the state of the petitioner to search cases by
+ * @param {string} providers.yearFiledMax the max year filed to search cases by
+ * @param {string} providers.yearFiledMin the min year filed to search cases by
```
Comment (kkoskelin) [inline shared/src/business/useCases/casePublicSearchInteractor.js]: These can probably be removed from the JSDOC on this function

File: shared/src/business/utilities/aggregateCommonQueryParams.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
+const { CaseSearch } = require('../entities/cases/CaseSearch');
+const aggregateCommonQueryParams = ({
```
Comment (kkoskelin) [inline shared/src/business/utilities/aggregateCommonQueryParams.js:15]: Could drop that bit of JSDOC here, instead.

**Review by kkoskelin — DISMISSED** [review]
Pretty neato. A few observations.

File: shared/src/business/useCases/casePublicSearchInteractor.js (hunk 2 above)
Comment (matthopson) [reply shared/src/business/useCases/casePublicSearchInteractor.js]: Just curious, why?

Comment (kkoskelin) [reply shared/src/business/useCases/casePublicSearchInteractor.js]: from what I can see, `countryType` and others are never referenced here in this function or explicitly named in the destructured args.

Comment (matthopson) [reply shared/src/business/useCases/casePublicSearchInteractor.js]: They aren't, but isn't it better to document the expected input object? I'm not doing anything different than we normally do except not explicitly destructuring the `provders` param. We do this on every function basically. I don't even think JSDOCS _know_ about the restructured params we normally document.

Comment (kkoskelin) [reply shared/src/business/useCases/casePublicSearchInteractor.js]: I thought this is what JSDOC was for -- to explain the arguments referenced within.  But have it your way.

**Pull request merged**